File: FishBun/src/main/java/com/sangcomz/fishbun/CustomizationProperty.java
package com.sangcomz.fishbun;

import android.graphics.drawable.Drawable;

/**
 * Created by sangcomz on 13/05/2017.
 */

interface CustomizationProperty {
    FishBunCreator setAlbumThumbnailSize(int size);

    FishBunCreator setPickerSpanCount(int spanCount);

    FishBunCreator setActionBarColor(int actionbarColor);

    FishBunCreator setActionBarTitleColor(int actionbarTitleColor);

    FishBunCreator setActionBarColor(int actionbarColor, int statusBarColor);

    FishBunCreator setActionBarColor(int actionbarColor, int statusBarColor, boolean isStatusBarLight);

    FishBunCreator setCamera(boolean isCamera);

    FishBunCreator textOnNothingSelected(String message);

    FishBunCreator textOnImagesSelectionLimitReached(String message);

    FishBunCreator setButtonInAlbumActivity(boolean isButton);

    FishBunCreator setAlbumSpanCount(int portraitSpanCount, int landscapeSpanCount);

    FishBunCreator setAlbumSpanCountOnlyLandscape(int landscapeSpanCount);

    FishBunCreator setAlbumSpanCountOnlPortrait(int portraitSpanCount);

    FishBunCreator setAllViewTitle(String allViewTitle);

    FishBunCreator setActionBarTitle(String actionBarTitle);

    FishBunCreator setHomeAsUpIndicatorDrawable(Drawable icon);

    FishBunCreator setOkButtonDrawable(Drawable icon);

    FishBunCreator setMenuText(String text);

    FishBunCreator setMenuTextColor(int color);

    FishBunCreator setIsUseDetailView(boolean isUse);
}

File: FishBun/src/main/java/com/sangcomz/fishbun/CustomizationProperty.kt
package com.sangcomz.fishbun

import android.graphics.drawable.Drawable

/**
 * Created by sangcomz on 13/05/2017.
 */
interface CustomizationProperty {
    fun setAlbumThumbnailSize(size: Int): FishBunCreator

    fun setPickerSpanCount(spanCount: Int): FishBunCreator

    fun setActionBarColor(actionbarColor: Int): FishBunCreator

    fun setActionBarTitleColor(actionbarTitleColor: Int): FishBunCreator

    fun setActionBarColor(actionbarColor: Int, statusBarColor: Int): FishBunCreator

    fun setActionBarColor(actionbarColor: Int, statusBarColor: Int, isStatusBarLight: Boolean): FishBunCreator

    fun setCamera(isCamera: Boolean): FishBunCreator

    fun textOnNothingSelected(message: String): FishBunCreator

    fun textOnImagesSelectionLimitReached(message: String): FishBunCreator

    fun setButtonInAlbumActivity(isButton: Boolean): FishBunCreator

    fun setAlbumSpanCount(portraitSpanCount: Int, landscapeSpanCount: Int): FishBunCreator

    fun setAlbumSpanCountOnlyLandscape(landscapeSpanCount: Int): FishBunCreator

    fun setAlbumSpanCountOnlPortrait(portraitSpanCount: Int): FishBunCreator

    fun setAllViewTitle(allViewTitle: String): FishBunCreator

    fun setActionBarTitle(actionBarTitle: String): FishBunCreator

    fun setHomeAsUpIndicatorDrawable(icon: Drawable?): FishBunCreator

    fun setDoneButtonDrawable(icon: Drawable?): FishBunCreator

    fun setAllDoneButtonDrawable(icon: Drawable?): FishBunCreator

    fun setIsUseAllDoneButton(isUse: Boolean): FishBunCreator

    fun setMenuDoneText(text: String?): FishBunCreator

    fun setMenuAllDoneText(text: String?): FishBunCreator

    fun setMenuTextColor(color: Int): FishBunCreator

    fun setIsUseDetailView(isUse: Boolean): FishBunCreator

    fun setIsShowCount(isShow: Boolean): FishBunCreator

    fun setSelectCircleStrokeColor(strokeColor: Int): FishBunCreator

    fun isStartInAllView(isStartInAllView: Boolean): FishBunCreator

    fun setSpecifyFolderList(specifyFolderList: List<String>): FishBunCreator

    fun hasCameraInPickerPage(isCamera: Boolean): FishBunCreator
}